### PR TITLE
Adding gated test for client telemetry service validation

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR contains 
1. Adding gated test for client telemetry service validation    
 (Juno stage endpoint will be available via environment variable added in tests.yml as cosmos-client-telemetry-endpoint 
 whose value is coming from our TestSecrets-Cosmos key vault)

2. Adding our default httptimeout for client telemetry request
3. refactored the package to remove camel case from package name com.azure.cosmos.implementation.client**T**elemetry -> com.azure.cosmos.implementation.clienttelemetry.

### CODE REVIEW

File: sdk/cosmos/tests.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ trigger: none
 variables:
   - template: ../../eng/pipelines/templates/variables/globals.yml
   - name: AdditionalArgs
-    value: ''
+    value: '-DCOSMOS.CLIENT_TELEMETRY_ENDPOINT=$(cosmos-client-telemetry-endpoint)'
```
Comment (simplynaveen20) [inline sdk/cosmos/tests.yml:6]: @benbp as discussed added the environment variable

**Review by moderakh — APPROVED** [review]
LGTM.

**Review by kushagraThapar — APPROVED** [review]
LGTM

**Pull request merged**